e2e/docker/docker.go

```diff
@@ -1157,6 +1193,60 @@ func (c ctx) testDockerCMDQuotes(t *testing.T) {
 // Check that the USER in a docker container is honored under --oci mode
 func (c ctx) testDockerUSER(t *testing.T) {
 	dockerURI := "docker://sylabsio/docker-user"
+	dockerfile := filepath.Join("..", "test", "defs", "Dockerfile.customuser")
+	tmpdir, tmpdirCleanup := e2e.MakeTempDir(t, "", "dockerfile-build-USER-", "temp dir for OCI-SIF images")
+	t.Cleanup(func() {
+		if !t.Failed() {
+			tmpdirCleanup(t)
+		}
+	})
+	userBuiltOCISIF := filepath.Join(tmpdir, "docker-user.oci.sif")
+	c.env.RunSingularity(
+		t,
+		e2e.AsSubtest("df build"),
+		e2e.WithProfile(e2e.UserProfile),
+		e2e.WithCommand("build"),
+		e2e.WithArgs("--oci", userBuiltOCISIF, dockerfile),
+		e2e.ExpectExit(0),
+	)
+	rootBuiltOCISIF := filepath.Join(tmpdir, "rootbuilt-docker-user.oci.sif")
+	c.env.RunSingularity(
+		t,
+		e2e.AsSubtest("df build"),
+		e2e.WithProfile(e2e.RootProfile),
+		e2e.WithCommand("build"),
+		e2e.WithArgs("--oci", rootBuiltOCISIF, dockerfile),
+		e2e.ExpectExit(0),
+	)
+
+	// Sanity check singularity native engine... no support for USER
+	c.env.RunSingularity(
+		t,
+		e2e.AsSubtest("default"),
+		e2e.WithProfile(e2e.UserProfile),
+		e2e.WithCommand("run"),
+		e2e.WithArgs(dockerURI),
+		e2e.ExpectExit(0, e2e.ExpectOutput(e2e.ContainMatch, fmt.Sprintf("uid=%d(%s) gid=%d",
+			e2e.UserProfile.ContainerUser(t).UID,
+			e2e.UserProfile.ContainerUser(t).Name,
+			e2e.UserProfile.ContainerUser(t).GID,
+		))),
+	)
+
+	metaTests := map[string]string{
+		"uri":  dockerURI,
+		"ubos": userBuiltOCISIF,
+		"rbos": rootBuiltOCISIF,
+	}
+
+	for subtestName, whatToRun := range metaTests {
+		t.Run(subtestName, func(t *testing.T) {
+			c.testDockerUSERWorker(t, whatToRun)
+		})
+	}
+}
+
+func (c ctx) testDockerUSERWorker(t *testing.T, whatToRun string) {
```

Comment:
`whatToRun` is a bit ambiguous... consider e.g. whether `what` is the container I'm running stuff in, or `what` could be what to run inside a container.

I'd probably go for `container` or something like that.